### PR TITLE
Categories fetching and category create api integrated

### PR DESCRIPTION
**Pull request by zubaerSumon**
modified categoy model, interface, validation
CRUD implemented for categories

### CODE REVIEW

File: src/server/modules/categories/categories.validation.ts
Comment (ronybarua-ethical18) [inline src/server/modules/categories/categories.validation.ts:1]: creator_id should be taken from api level. The field should be removed from categories validation.

File: src/server/modules/categories/index.ts
Comment (ronybarua-ethical18) [inline src/server/modules/categories/index.ts:1]: The async code blocks should be wrapped up with a try-catch block.

Comment (ronybarua-ethical18) [reply src/server/modules/categories/index.ts:1]: The existence of the category should be checked with the creator_id field also.

File: src/server/db/models/category.ts
Comment (ronybarua-ethical18) [inline src/server/db/models/category.ts:1]: add indexing for title in categories model.

**Review by ronybarua-ethical18 — COMMENTED** [review]
Follow the feedbacks provided for category module and push again with the final code.

File: src/server/modules/categories/categories.validation.ts
Comment (zubaerSumon) [reply src/server/modules/categories/categories.validation.ts:1]: field creator id is red marked indicating removed before

**Pull request merged**